q2_cutadapt/_demux.py

```diff
@@ -219,9 +250,8 @@ def demux_paired(seqs: MultiplexedPairedEndBarcodeInSequenceDirFmt,
                  cores: int = 1) -> \
                     (CasavaOneEightSingleLanePerSampleDirFmt,
                      MultiplexedPairedEndBarcodeInSequenceDirFmt):
-    if mixed_orientation and reverse_barcodes is not None:
```

Comment:
Hey @vaamb you're seeing a test failure ([here](https://github.com/qiime2/q2-cutadapt/actions/runs/4499221194/jobs/7916815093?pr=55#step:4:753)) because this error branch has been removed.

Reply:
Thanks, I'll have a look. But first I need to go through the dev docs to be able to run the tests locally; somehow, I can’t make them work.

Reply:
@vaamb you should be able to run `pytest` while sitting in the top-level directory of q2-cutadapt to run these unit tests locally. Let us know if there's anything specific that you're stuck on while trying to test!

Reply:
@vaamb can you provide a screenshot of the entire command you're running, along with the full traceback with the test failures?

Reply:
@lizgehret I was launching `pytest` via Pycharm and somehow it messed things up, by simply running it in the terminal I made it work!

I modified one existing test to test if some "barcode1-barcode2" and "barcode2-barcode1" pairs are present and it works as expected.

However, I have recently found an edge case ("barcode1-barcode1") that technically is correct for demultiplexing but can lead to some downstream issues (as both sequences and their reverse complement are considered as distinct). 
One solution is to allow it and advise to do a de novo clustering using vsearch with the strand parameter set to both (I made a pull request at qiime2/q2-vsearch#90 to expose it). That is what I currently implemented using `print()` (is there any other mean to log such a warning?)
The other solution would be to raise an error before